filters/ratelimit/ratelimit.go

```diff
@@ -145,25 +161,43 @@ func clusterRatelimitFilter(args []interface{}) (filters.Filter, error) {
 		Type:       ratelimit.ClusterServiceRatelimit,
 		MaxHits:    maxHits,
 		TimeWindow: timeWindow,
+		Lookuper:   ratelimit.NewSameBucketLookuper(),
+	}
+
+	return &filter{settings: s}, nil
+}
+
+func clusterClientRatelimitFilter(args []interface{}) (filters.Filter, error) {
+	if !(len(args) == 2 || len(args) == 3) {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	maxHits, err := getIntArg(args[0])
+	if err != nil {
+		return nil, err
+	}
+
+	timeWindow, err := getDurationArg(args[1])
+	if err != nil {
+		return nil, err
+	}
+
+	s := ratelimit.Settings{
+		Type:       ratelimit.ClusterClientRatelimit,
+		MaxHits:    maxHits,
+		TimeWindow: timeWindow,
 	}
 
 	if len(args) > 2 {
-		lookuperName, err := getStringArg(args[2])
+		headerName, err := getStringArg(args[2])
 		if err != nil {
 			return nil, err
 		}
-		switch lookuperName {
-		case "auth":
-			s.Lookuper = ratelimit.NewAuthLookuper()
-			s.CleanInterval = 10 * timeWindow
-		case "xfwd":
-			s.Lookuper = ratelimit.NewXForwardedForLookuper()
-			s.CleanInterval = 10 * timeWindow
-		default:
-			return nil, filters.ErrInvalidFilterParameters
-		}
+		s.Lookuper = ratelimit.NewHeaderLookuper(headerName)
+		s.CleanInterval = 10 * timeWindow
 	} else {
-		s.Lookuper = ratelimit.NewSameBucketLookuper()
+		s.Lookuper = ratelimit.NewXForwardedForLookuper()
```

Comment:
does this mean that when we specify the only the time window and the max requests, then we will always use the X-Forwarded-For header for bucketing by default? This is not the same as the previous behavior.

Reply:
yes

Reply:
It looks like this is already merged, but as far as I understand in current implementation if have `clusterClientRatelimit(20, "1h", "X-Forwarded-For")` it is different from not providing a header.

Also probably makes sense to either normalize header casing names unless I'm missing this part.

Reply:
Please create an issue for the normalizing and yes the header thing is different from the default.